server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1897,11 +1913,6 @@ private static SnapshotDeletionsInProgress deletionsWithoutSnapshots(
         return changed ? SnapshotDeletionsInProgress.of(updatedEntries) : null;
     }
 
-    private void failSnapshotCompletionListeners(Snapshot snapshot, Exception e) {
```

Comment:
removed `failSnapshotCompletionListeners ` method as it is a "collection of side-effects". inlined its logic for better readability

Reply:
I'd prefer to keep this in its own method, it's used in a lot of different places that now have duplicated code which I expect will drift apart over time now.

Reply:
I see, brought method back and overloaded it with extra param - failing listeners collector